### PR TITLE
Fixes #12530 - Upgrade rubocop to 0.35.1

### PR DESCRIPTION
**Pull request by daviddavis**
This is mostly because of improved problem detection. A couple cops have bugs that cause false positives and negatives. There are also some new cops I'd like to enable like `Lint/DuplicateMethods` but will do so after we upgrade to Rails 4

### CODE REVIEW

**Comment by domcleal** [comment]
Are the disabled cops all of the new ones, or just all of the new ones that fail in some way?  (Missing offence counts.)

It might be preferable to refresh the todo file from --auto-gen-config, so we get to use the new cops for future changes and keep them disabled only where there are existing problems?  e.g. cops such as Performance/Detect are valuable.

**Comment by daviddavis** [comment]
These are just the new cops that are failing. In rubocop 0.33.0 they changed --auto-gen-config to exclude files that have problems instead of disabling the cops altogether (see https://github.com/fusor/fusor/pull/465/files). This effectively turns on the cop for any new code. I'm not sure I like that. 

However, it looks like I can generate the todo file with cops being disabled by using `--exclude-limit 0`. I've updated the PR with this so we can get the offense counts.

Let me know if you prefer excluding files over disabling outstanding cops.

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -1,88 +1,159 @@
...
 # Cop supports --auto-correct.
+# Configuration parameters: AllowUnusedKeywordArguments, IgnoreEmptyMethods.
 Lint/UnusedMethodArgument:
   Enabled: false
 
-# Offense count: 459
+# Offense count: 494
 Metrics/AbcSize:
-  Max: 194
+  Max: 114
```
Comment (dLobatog) [inline .rubocop_todo.yml:66]: Glad to see we decreased the max since Rubocop was added :smile:

**Comment by dLobatog** [comment]
@daviddavis Rubocop is complaining about the length of 2 classes (no kidding), I'd say :+1: after fixing that.

**Comment by daviddavis** [comment]
@dLobatog thanks. Should be fixed now.

**Comment by domcleal** [comment]
> Let me know if you prefer excluding files over disabling outstanding cops.

Yes, I did prefer the exclusion of files, so we get the benefit of some cops on new code - though I guess I was particularly interested in the performance ones, less that we should enforce different style.  Should we make that distinction?

**Comment by daviddavis** [comment]
@domcleal updated. thanks.

**Comment by domcleal** [comment]
Merged as a9d8cde, thanks @daviddavis.

**Pull request closed without merging**